boa/src/syntax/parser/cursor/buffered_lexer/mod.rs

```diff
@@ -82,6 +82,12 @@ where
         self.lexer.lex_slash_token(start).map_err(|e| e.into())
     }
 
+    /// Lexes the next tokens as template middle or template tail assuming that the starting
+    /// '}' has already been consumed.
+    pub(super) fn lex_template(&mut self, start: Position) -> Result<Token, ParseError> {
+        self.lexer.lex_template(start).map_err(|e| e.into())
```

Comment:
Could this be replaced by this?
```suggestion
        self.lexer.lex_template(start).map_err(ParseError::from)
```

It would remove the need for the closure.